src/serac/physics/solid_functional.hpp

```diff
@@ -368,17 +367,16 @@ class SolidFunctional<order, dim, Parameters<parameter_space...>, std::integer_s
             deformation_grad = deformation_grad + du_dX;
           }
 
-          // Note that the jacobian needs the fixup for the shape displacement.
-          // The material returns Kirchoff stress, which contains det(du / dX'). We want
-          // The final Jacobian to be det(du / dX), so we compute
-          // det(du / dX) = det(du / dX') * det(dX' / dX) = det(du / dX') * det(I + dp / dX)
-          auto flux = dot(stress, transpose(inv(deformation_grad))) * det(I + dp_dX);
+          auto flux = dot(stress, transpose(inv(deformation_grad))) * det(deformation_grad);
```

Comment:
Don't we still need to multiply by `det(I + dp_dX)`? I agree that the `det(deformation_grad)` used to be absorbed in the stress, but now it isn't, so we have to multiply by it. But I don't see why we get to drop the `det(I + dp_dX)` factor.

Reply:
We don't. This his now included in `det(deformation_grad) = det(dx_dX') * det(dX'_dX) =  det(dx_dX') * det(I + dp_dX)`.

Reply:
Ok, from our verbal conversation, I got the piece I was missing: `deformation_grad` is not the "real" deformation gradient for the problem $\frac{\partial x}{\partial X'}$, but the gradient with respect to the original reference configuration coordinates $X$.